### PR TITLE
Script for feature "Remote Control - Update OnRCStatus with a new allowed parameter"

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[2244](https://github.com/smartdevicelink/sdl_core/issues/2244)
This PR is **ready** for review.

### Summary
Script for feature "Remote Control - Update OnRCStatus with a new allowed parameter"

### ATF version
[Develop branch](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/RC/OnRCStatus/001_notification_by_apps_registration.lua [hunk 1]
```diff
@@ -9,7 +9,7 @@
 -- In case:
 -- 1) RC app1 and RC app2 are registered
 -- SDL must:
--- 1) Send OnRCStatus notifications to all registered mobile applications and the HMI
+-- 1) Send OnRCStatus notifications to all registered mobile applications and the HMI with allowed = true
```
Comment (yang1070) [inline test_scripts/RC/OnRCStatus/001_notification_by_apps_registration.lua]: a RC app register event shall trigger an OnRCStatus notification sent to the newly registered RC app only, not to existing RC apps or non-RC apps, as there is no change to the array of freeModules and allocatedModules

File: test_scripts/RC/OnRCStatus/002_notification_by_rc_functionality_disallowed_on_hmi.lua [hunk 2]
```diff
@@ -10,7 +10,7 @@
 -- 1) RC functionality is disallowed on HMI
 -- 2) RC application is registered
 -- SDL must:
--- 1) Not send OnRCStatus notification to registered mobile application and the HMI
+-- 1) Send OnRCStatus(allowed = false, freeModules = {}, allocatedModules = {}) notification to registered mobile application and the HMI
```
Comment (yang1070) [inline test_scripts/RC/OnRCStatus/002_notification_by_rc_functionality_disallowed_on_hmi.lua]: why send to HMI?
Send OnRCStatus(allowed = false, freeModules = {}, allocatedModules = {}) notification to registered mobile `RC` applications

**Comment by GetmanetsIrina** [comment]
@yang1070 , please find all updates in last commit.

**Comment by yang1070** [comment]
@theresalech Ford reviewed and approve this PR.

**Pull request merged**